src/Catel.MVVM/Catel.MVVM.Shared/MVVM/ViewModels/ViewModelBase.cs

```diff
@@ -543,13 +544,13 @@ private void InitializePropertiesWithAttributes()
         /// <param name="viewModelType">Type of the view model.</param>
         /// <returns>ViewModelMetadata.</returns>
         /// <exception cref="ArgumentNullException">The <paramref name="viewModelType" /> is <c>null</c>.</exception>
-        private static ViewModelMetadata InitializeViewModelMetaData(Type viewModelType)
+        private static ViewModelMetadata GetViewModelMetaData(Type viewModelType)
         {
-            if (_metaData.ContainsKey(viewModelType))
-            {
-                return _metaData[viewModelType];
-            }
+            return _metaData.GetOrAdd(viewModelType, CreateViewModelMetaData);
```

Comment:
Note that without locking a concurrent dictionary, it is possible that *CreateViewModelMetaDAta* is called multiple times (but will only be stored once).